hlf/scala/general-communication.md

```diff
@@ -65,35 +65,56 @@ The ADMIN can then
+- have the USER create their signature for the "unsigned" proposalBytes
 - receive the signature from the USER (together with the original proposalBytes)
 
-- submit the signed proposal ("unsigned" proposalBytes, signatureBytes) to retrieve an usigned Transaction
+- submit the signed proposal ("unsigned" proposalBytes, signatureBytes) to retrieve an usigned Transaction.
+> While the proposal is fetched from the specific connections (e.g. MatriculationConnection, CertificateConnection, ...), the getUnsignedTransaction, which submits the signed proposal, is called on the OperationConnection.
 ```scala
-val transactionBytes = certificateConnection.getUnsignedTransaction(proposalBytes: Array[Byte], signatureBytes: Array[Byte])
+val transactionBytes = operationConnection.getUnsignedTransaction(proposalBytes: Array[Byte], signatureBytes: Array[Byte])
 ```
 
 - pass the transactionBytes on to the USER who wants to sign it
 - have the USER create his signature for the "unsigned" transactionBytes
 - receive the signature from the USER (together with the original transactionBytes)
 
 - submit the signed transaction ("unsigned" transactionBytes, signatureBytes)
+>The Connection receiving the signedTransaction is always the OperationConnection.
 ```scala
-val (approvalResult, realTransactionResult) = certificateConnection.submitSignedTransaction(transactionBytes: Array[Byte], signatureBytes: Array[Byte])
+val approvalResult = operationConnection.submitSignedTransaction(transactionBytes: Array[Byte], signatureBytes: Array[Byte])
+
+val transactionResult = operationConnection.executeTransaction(jsonOperationData: String, timeoutMilliseconds: int = 5000)
 ```
-The approvalResult is a JsonObject [OperationData](../chaincode/operation.md#OperationData).
-The realTransactionResult is the chaincode response to the desired transaction
-(e.g. an InsufficientApproval Error, or information about the successfully changed ledger state)
+The approvalResult is a JsonObject [OperationData](../chaincode/operation.md#OperationData). 
+Invoking executeTransactionFromOperation executes the transaction described by the transactionInfo in the OperationData.
+In case of a NetworkException, the transaction is retried until the specified timeout is reached.
+The transactionResult is the chaincode response to the desired transaction
+(e.g. an InsufficientApproval Error, or information about the successfully changed ledger state).
 
-The approval of the User has been stored on the ApprovalContract.
-An attempt has been made to execute the "real" transaction (e.g. "AddCertificate")
+If submitSignedTransaction succeeds, the approval of the User has been stored on the OperationContract.  
+If executeTransaction succeeds, the "real" transaction (e.g. "AddCertificate") is executed.  
+If the approval or the "real" transaction fails due to the transaction itself being invalid (e.g. due to changed ledger state), the operation is rejected.
```

Comment:
rejected by? the system? the chain? what enrollmentId/grou?

Reply:
```suggestion
If the approval or the "real" transaction fails due to the transaction itself being invalid (e.g. due to changed ledger state), the operation is rejected by the chain.
```